node/src/reactor/participating.rs

```diff
@@ -1113,6 +1120,18 @@ impl reactor::Reactor for Reactor {
 
                 effects
             }
+            ParticipatingEvent::BlockProposerAnnouncement(
+                BlockProposerAnnouncement::DeploysExpired(hashes),
+            ) => hashes
+                .into_iter()
+                .map(|hash| {
+                    let reactor_event = ParticipatingEvent::EventStreamServer(
+                        event_stream_server::Event::DeployExpired(hash),
+                    );
+                    self.dispatch_event(effect_builder, rng, reactor_event)
+                })
+                .flatten()
+                .collect(),
```

Comment:
So, I am wondering if its possible to have the event stream server, do the flattening of the deploy hashes on its side. Here I feel like we're pumping the event queue with a bunch of events, whereas I feel like we should have the Event server collect all the hashes and then handle the individual hashes on its part of the component boundary. 

@Fraser999 thoughts???

Reply:
@Fraser999 
Although we initially agreed that we can live with the current implementation, I'm keen to change it, given we have the comment from @darthsiroftardis .

It all boils down to how many such events we expect to get and if it's better to have a single, potentially big event or several smaller ones.

Reply:
Yes, I agree with Karan.

We want to minimise the work being done in `dispatch_event` and offload it to the event stream server.

Reply:
Fixed.
The participating reactor now flushes a vector of expired hashes to the event stream server, which then creates appropriate events on its own.